File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.cpp
class Solution {
public:
    string decodeCiphertext(string encodedText, int rows) {
        
        if( encodedText == "" ) return encodedText ;
        string result = "" ;

        int col = 0 ;
        int n = encodedText.size() / rows ;
        bool valid = true ;

        while( valid && result.size() != encodedText.size() ){

            int colStart = col ;
            
            for( int i = 0 ; i < rows ; i++ ){
                int index = ( i * n ) + colStart ;
                if( index < encodedText.size() ){
                    result.push_back( encodedText[index] );
                    colStart ++ ;
                }
                else{
                    valid = false ;
                }
            }
            col ++ ;
        }

        while( result.back() == ' ' ) result.pop_back() ;

        return result ;
    }
};

File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.js
/**
 * @param {string} encodedText
 * @param {number} rows
 * @return {string}
 */
var decodeCiphertext = function(encodedText, rows) {
    
    if( encodedText.length == "" ) return "" ;

    let n = Number(encodedText.length / rows) ;
    let valid = true ;
    let col = 0 ;
    let result = "" ;

    while( valid && result.length != encodedText.length ){

        let colStart = col ;
        for( let i = 0 ; i < rows ; i++ ){

            let index = ( i * n ) + colStart ;
            if( index < encodedText.length ){
                result += encodedText[index] ;
            }
            else{
                valid = false ;
            }
            colStart ++ ;
        }
        col ++ ;
    }

    while (result.length > 0 && result[result.length - 1] === ' ') {
        result = result.trimEnd();
    }

    return result ;

};